### PR TITLE
Travis config

### PR DESCRIPTION
**Pull request by nbrl**
Added a travis config so tests should automatically run against Sauce Labs using Travis. Also updated the dev section of the README.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -248,6 +248,14 @@ TODO: do you think we should document the 'under the hood' props, e.g. `clientTo
 
 ## Development
 
+### Set up
+To set up your local environment, clone this repo and from within it run:
+```
+npm install   # installs development dependencies
+bower install # installs frontend dependencies
+gulp build    # builds front end dependencies into distributable js
```
Comment (jabclab) [inline README.md]: Does `gulp` need to be installed globally à la Grunt?

File: package.json [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {
-  "name": "build-things-js-sdk",
+  "name": "advocate-things-js-sdk",
```
Comment (jabclab) [inline package.json]: There are some more references below too

File: README.md [hunk 3]
```diff
@@ -8,7 +8,7 @@ The official Advocate Things SDK for JavaScript (currently available for browser
 In order to use the SDK in the browser, simply add the following to your HTML pages:
 
 ```html
-<script src="https://sdk.at.com/at-sdk-0.0.1.js?key=YOUR_KEY"
+<script id="advocate-things-script" src="https://advocate-things.divshot.com/at-sdk-0.0.1.js?key=YOUR_KEY"
```
Comment (jabclab) [inline README.md]: Annoyingly the introduction of the `id` attribute makes this line longer than the default GitHub README.md screen. Maybe move `src` onto the next line?

File: README.md [hunk 4]
```diff
@@ -248,6 +248,15 @@ TODO: do you think we should document the 'under the hood' props, e.g. `clientTo
 
 ## Development
 
+### Set up
+To set up your local environment, clone this repo and from within it run:
+```
+npm -g install gulp # installs gulp globally for js file building
```
Comment (jabclab) [inline README.md]: `$` for the start of each of these (plus maybe `[sudo]` before the `npm install -g ...`).

File: README.md [hunk 5]
```diff
@@ -255,11 +264,18 @@ In order to run tests simply execute:
...
+automatically re-running the tests when necessary. When doing this, it is also
+necessary to run `gulp watch` to ensure the built file is up to date before
+testing.
 
 ### Building
 In order to build the JS file use the following:
 
 ```
-$ ./build-script.sh
+gulp build
```
Comment (jabclab) [inline README.md]: `$`

File: README.md (hunk 3 above)
Comment (nbrl) [inline README.md]: Would that not stop one being able to simply copy and paste it?

Comment (jabclab) [inline README.md]: Shouldn't do as it's already on two lines :smile:

File: README.md (hunk 4 above)
Comment (nbrl) [inline README.md]: Commented that you may need `sudo`, but won't say to run it with `sudo` as then people will run it with `sudo` without thinking.

**Pull request merged**